### PR TITLE
[IMP] website: make contactus page static

### PR DESCRIPTION
**Pull request by tsm-odoo**
Currently, contactus page is dynamic (Google map snippet, Company Data, Contact Form), in order to make it static : 
- Company data snippet has been removed and replace by static data
- Google map snippet has been removed
- Website_form module has been merged into website (before, the module needed to add data into contact us page which was not static)
- Pre-fill option has been added  to s_website_form snippet and all website_form dependencies have been updated to  add "fillWith" attribute during form model registry.
- Finally, contactus-thanks has also become static !

Enterprise: https://github.com/odoo/enterprise/pull/18242
Upgrade: https://github.com/odoo/upgrade/pull/2452
task-2462993

### CODE REVIEW

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:138]: remove the `mt-4 mb-5`, this is creating a double spacing as there is already a spacing on the line above (section pt40 & pb40) + margin can't be changed by editor

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:15]: remove the `mt-4 mb-5`, this is creating a double spacing as there is already a spacing on the line above (section pt40 & pb40) + margin can't be changed by editor

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:110]: `<address>` is now useless since we don't use t-fields + "widget": "contact" anymore, which were in charge of adding the itemprop :/
So either we remove the `<address>` schema tag or we add the itemprop in the DOM (including on phone and on mail, but those are not supposed to be inside address tag AFAIK so it seems it was wrong before).

One problem would be that the enduser would mess with those hardcoded itemprop, eg replace the phone by the email and now the itemprop phone contains an email..

So I'd vote to simply remove the `<address>` since it is not a t-field anymore and it makes sense only for a t-field as we know what will be written inside.

Anyone else's opinion is welcome here
@JKE-be @sbuhl

File: addons/website/views/website_form_templates.xml
Comment (rdeodoo) [inline addons/website/views/website_form_templates.xml]: Why were those oe_structure IDs element removed? The templates are moved to website but those are simply gone.
Is you blamed those line, you see it is there on purpose:
Added everywhere with (among other commits): https://github.com/odoo/odoo/commit/4f15c939a78b4f1f20377d62e89157dfe628d910
In order to achieve this: https://github.com/odoo/odoo/commit/dd139948f0049b3676b859495ba9b2d43436f9c1

Was it discussed with someone or just a mistake to remove those?

File: addons/website/views/website_form_templates.xml
Comment (rdeodoo) [inline addons/website/views/website_form_templates.xml]: same here about oe_structure IDs

File: addons/website_crm/views/website_crm_templates.xml
Comment (rdeodoo) [inline addons/website_crm/views/website_crm_templates.xml]: I am not too sure about that, does that mean when `website_crm` module is installed, there won't be any lead created when contact form is submitted :thinking: ?
That's the whole point of the module. Maybe I am missing something here :)

**Review by rdeodoo — CHANGES_REQUESTED** [review]
It is almost impossible to review the contactus template and contactus thank you page changes, as there is a move + a diff in a single commit.
May I ask you to create a move commit where you move:
- _website.contactus template (from /website/data/website_data.xml to /website/data/website_contactus_data.xml)
- _website.contactus_form_ template (from /website/views/website_form_templates.xml to /website/data/website_contactus_data.xml)
- _website.contactus_thanks template (from /website/views/website_form_templates.xml to /website/data/website_contactus_data.xml)

And do the actual changed in a later commit?
Right now we have to copy paste the code before & after on a text compare tool to see the actual changes, and we will have to do that after every PR change, not ideal

After that, I will be able to do an real review

File: addons/website_crm/views/website_crm_templates.xml
Comment (rdeodoo) [reply addons/website_crm/views/website_crm_templates.xml]: @sbuhl ?

Comment (rdeodoo) [inline addons/website_crm/views/website_crm_templates.xml]: Without this, lead are not generated anymore when form is submitted, isn't it?
That's the whole purpose of the module, unless I miss something obvious here it seems strange?

File: addons/website/views/website_form_templates.xml
Comment (tsm-odoo) [reply addons/website/views/website_form_templates.xml]: I think it was qsm meant in my other contactus PR ? What do you think ?

> I don't think there is a reason for the "thank you" page to not be fully static too, confirm that with @sbuhl. But in that case, there won't be two oe_structure around the page content and the customization of those in current user database would have to be migrated.

>  Le but c'est que ces pages "contactus", "thank you" etc soient exactement comme des pages que l'utilisateur aurait créées lui-meme.

If I drag and drop snippets to create a page, there only will be an `oe_structure` on the `#wrap div`

File: addons/website_crm/views/website_crm_templates.xml
Comment (tsm-odoo) [reply addons/website_crm/views/website_crm_templates.xml]: They are but we now have to change the form action first. I did add a step to the website_crm tour we already had to reflect this change, and everything is working as it was before except for the fact we need to enable `Create an opportunity` action.

Comment (rdeodoo) [reply addons/website_crm/views/website_crm_templates.xml]: Yes but that doesn't seems acceptable to me, the module is not doing what it is supposed to do, unless the enduser actually enter edit mode and change a specific option, which enduser will never, never, never do.

File: addons/website/views/website_form_templates.xml
Comment (rdeodoo) [reply addons/website/views/website_form_templates.xml]: Ok so it was discussed and on purpose :+1: 
Not sure why there was a PR change and those review/comments are lost but fine

Comment (JKE-be) [reply addons/website/views/website_form_templates.xml]: > I don't think there is a reason for the "thank you" page to not be fully static too

Really sad to open a ticket and don't have an reference on the thank you page :/

File: addons/website_crm/views/website_crm_templates.xml
Comment (rdeodoo) [reply addons/website_crm/views/website_crm_templates.xml]: IMO, on module install post init hook, we should at least write on the website.contactus page(s) to change `data-model_name` to `crm.lead` so the module is "ready to work" as expected and as it is without this task.

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:147]: useless diff (+ we don't normally add space before auto closing tags)

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:144]: useless diff (+ we don't normally add space before auto closing tags)

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:141]: No need to change but for futur pull request, this is not a mandatory change, as it just creates 2 lines of diff for no real reason
This diff is already a bit hard to analyze (even with hide whitespace changes) so this is the kind of change to avoid :)

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:137]:
```suggestion
                            <div class="container s_allow_columns">
```

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:154]:
```suggestion
                                        <ul class="list-unstyled mb-0 pl-2">
                                             <li>My Company</li>
```
Diff is weird but I'd just move the My Company from the `<p>` to a new `<li>`, so:
1. We don't need those "I use a `<p>` but I don't want the `<p>` style" classes (p-0 m-0)
2. It will keep the same layout/design as before without an extra indent visually

See 
image
First "My Company" is in a `<p>` while second is in `<li>`

File: addons/website_sale/static/src/js/website_sale_form_editor.js
Comment (rdeodoo) [inline addons/website_sale/static/src/js/website_sale_form_editor.js:11]: Commit title is `[IMP] website : add pre fill option for website form` but it is also modifying other modules than website, needs to be indacted too.
Also remove the whitespace between `website :`

File: addons/website/static/src/snippets/s_website_form/000.js
Comment (rdeodoo) [inline addons/website/static/src/snippets/s_website_form/000.js:66]: Shouldn't this list of field be inside a function?
Imagine we want to add a website_sale related field (not existing in website) inside the form later (or the community), this module would need to be able to overide that list no?
That would be enough for it to work, right? Add field + data-fill-with in FormRegistry then txtend this list to read that field?

**Review by rdeodoo — CHANGES_REQUESTED** [review]
Then we should be ready to go :)
Can you rewrite every commit title to remove the extra space between module and : ? (+ ensure modified modules are all listed)
Also as we will probably use a merge commit here, can you add a message to the PR description?

Edit: I don't agree with the spec to remove the "ready to work" capacility of the `website_crm` module, I am pretty sure no one will understand they have to edit the form to change the action after installing the module.. But that's not up to me to decide..

File: addons/website_crm/views/website_crm_templates.xml
Comment (qsm-odoo) [reply addons/website_crm/views/website_crm_templates.xml]: What does not seem acceptable for @rdeodoo here was normally decided with @sbuhl and @JKE-be (as written in the spec)... did you discuss with him or did you end up implementing the post init hook stuff @tsm-odoo ? I don't think it is a viable idea, you don't know if there is even a form in this contact us page :grimacing: 

Original goal: contact us page is static so no magic involved in what it does: this is a pre-made form for you to contact the company -> you want it to do/show other things? Edit the page.

We can change our mind of course but let's not enter an infinite loop of spec changes please :wink: @sbuhl as this part of the spec is dubious since the beginning maybe it's time to involve FP so this spec does not change again? We already changed our mind about it like 3 times before ^^"

Comment (tsm-odoo) [reply addons/website_crm/views/website_crm_templates.xml]: @qsm-odoo No it's not implemented, indeed there were some issues like the fact that all form models have different field names... We discussed it during last week meeting and @JKE-be  told us that it was indeed the specs. Still, @rdeodoo  thinks it's a bad idea :
> I don't agree with the spec to remove the "ready to work" capacility of the website_crm module, I am pretty sure no one will understand they have to edit the form to change the action after installing the module.. But that's not up to me to decide..

Anyway, @rdeodoo told me he was going to last review this task today... Let me know if something changes :)

Comment (qsm-odoo) [reply addons/website_crm/views/website_crm_templates.xml]: Ok :wink: Yes, @JKE-be and I had the same fears as @rdeodoo but, specs :wink: I actually see the value in having the contact us page static... and keeping the CRM "auto" feature seems impossible/complicated functionally and technically so compromise... :grimacing: 

IMO this is the kind of stuff our navbar warning task could have been about too :( "Hint: your contact us page uses a form which sends mails, what about creating opportunities?" cc @sbuhl :wink: Useful hint, warning and error reports >>>>>> useless tours that nobody follows (especially when it's only for admins...)

File: addons/website/__manifest__.py
Comment (rdeodoo) [inline addons/website/__manifest__.py:28]: alphabetical, move one line up

File: addons/website/__manifest__.py
Comment (rdeodoo) [inline addons/website/__manifest__.py]: alphabetical, move one line up

File: addons/website/__manifest__.py
Comment (rdeodoo) [inline addons/website/__manifest__.py:91]: alphabetical, move 3 lines bellow

File: addons/website/__manifest__.py
Comment (rdeodoo) [inline addons/website/__manifest__.py:207]: Shouldn't this be inside `website.assets_wysiwyg` asset?

File: addons/website/__manifest__.py
Comment (rdeodoo) [inline addons/website/__manifest__.py:206]: alphabetical, move before the /menu/ files

File: addons/website/models/__init__.py
Comment (rdeodoo) [inline addons/website/models/__init__.py:29]: alphabetical please, you can also reorder the last 3 before your line.. lets keep things clean

```
from . import website_configurator_feature
from . import website_form
from . import website_snippet_filter
from . import website_visitor
```

File: addons/website/tests/__init__.py
Comment (rdeodoo) [inline addons/website/tests/__init__.py:23]: :'( 
also reorder the last 2 before your line while we are on it..

File: addons/website_hr_recruitment/__manifest__.py
Comment (rdeodoo) [inline addons/website_hr_recruitment/__manifest__.py:11]: Why reordered this way? Reorder alphabetical if you do it :)

**Review by rdeodoo — CHANGES_REQUESTED** [review]
After this, @JKE-be or @qsm-odoo might have the final review before r+

**Comment by tsm-odoo** [comment]
@qsm-odoo @JKE-be, I'm done applying @rdeodoo suggestions, I'm waiting for your final review :)

File: addons/website/data/website_contactus_data.xml
Comment (qsm-odoo) [inline addons/website/data/website_contactus_data.xml]: In this commit, you move/rename a file (already renamed originally from website_form) and modify it at the same time. So forward-port and history etc might fail (this is the whole point of your first commit) -> rename it directly like in this third commit but in the first commit (so that the third commit is only about moving code inside).

Note: here GitHub understands this is still a move despite the configuration -> that might not be the case for all git configurations

File: addons/website/data/website_contactus_data.xml
Comment (qsm-odoo) [inline addons/website/data/website_contactus_data.xml:7]: Is not there an `oe_empty` class alongside the `oe_structure` class in default pages?

File: addons/website/data/website_contactus_data.xml
Comment (qsm-odoo) [inline addons/website/data/website_contactus_data.xml:135]: Same here

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [reply addons/website/data/website_contactus_data.xml]: If this comment is about `[MOV] website: move website contactus into separate file` commit, there is no change, isn't it?
It is just a move because later commit actually change the code.
I asked for this commit to avoid avoid the move in the next one, see https://github.com/odoo/odoo/pull/69888#pullrequestreview-661084659

Comment (qsm-odoo) [reply addons/website/data/website_contactus_data.xml]: In that commit: :wink: 

Screenshot from 2021-06-07 16-29-14

Comment (rdeodoo) [reply addons/website/data/website_contactus_data.xml]: Ok I see, I guess it is how github show "part move" instead of complete move :shrug: as there is not a single change/diff here

Comment (qsm-odoo) [reply addons/website/data/website_contactus_data.xml]: When I say "move", I am talking about renaming a file path ;) Moving code from a file to another is a "change".

File: addons/website_crm/views/website_crm_templates.xml
Comment (qsm-odoo) [inline addons/website_crm/views/website_crm_templates.xml]: TODO: update module description

**Review by qsm-odoo — COMMENTED** [review]
Notes for monday

Also, what about dependencies like website_helpdesk_form... should also be merged in website_helpdesk, etc

File: addons/website/__manifest__.py
Comment (rdeodoo) [reply addons/website/__manifest__.py:20]: mail was already in it through portal :) portal depends of mail

Comment (rdeodoo) [reply addons/website/__manifest__.py:20]: so is digest and auth_signup

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [inline addons/website/data/website_contactus_data.xml:1]: TODO: Probably also need to move this back inside website_data.xml where it came from
less diff + with other pages where is was
Or at least adapt the comment `<!-- Template for home and contactus -->` since contactus is not there anymore

File: addons/website/__manifest__.py
Comment (qsm-odoo) [reply addons/website/__manifest__.py:20]: Strange, I did test functionally to install website and checked if discuss was installed... I probably messed up my test x) (merging website_mail into website is still a possibility though but I guess we can ignore that here to move forward

File: addons/website_crm/views/website_crm_templates.xml
Comment (rdeodoo) [reply addons/website_crm/views/website_crm_templates.xml]: done

File: addons/website/data/website_contactus_data.xml
Comment (rdeodoo) [reply addons/website/data/website_contactus_data.xml]: Done

**Comment by rdeodoo** [comment]
> Also, what about dependencies like website_helpdesk_form... should also be merged in website_helpdesk, etc

@qsm-odoo I don't think it should be, it is not an auto install module but an opt-in feature that you might want to install (or not) on top of website_helpdesk

**Comment by qsm-odoo** [comment]
> > Also, what about dependencies like website_helpdesk_form... should also be merged in website_helpdesk, etc
> 
> @qsm-odoo I don't think it should be, it is not an auto install module but an opt-in feature that you might want to install (or not) on top of website_helpdesk

Indeed :+1:

**Comment by rdeodoo** [comment]
@qsm-odoo 

Everything that was discussed yesterday should've been done, plus some last minute improvements:
- fixed xml data files order, otherwise form field would not be whitelisted,
  including `email_to`, so the mail would not be send.
- add the same data-for behavior from website_crm to website, to be able to
  prefill a field with an URL param
- removed the fallback data-for in website_crm, the JS in now in charge of
  doing that. Plus it was creating a misheavior -> you could have prefill set
  to false but still, it would prefill the user info
- fixed the form option so the field.name is not used as field.string if there
  is already an explicitely set field.string..
- adapt field name in /contactus form (xml view), form snippet (registry) and
  crm form (registry) for more consistence
- added the prefill option set to true by default (to fit what we had before)
- changed the pre-fill/request.param order: param takes over pre-fill if set

Also updated the upgrade PR, tests moved in website_crm

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: html set with the content of rendered template, not injectable content

**Comment by JKE-be** [comment]
@robodoo delegate=rdeodoo
(I don't check again since you did the review with QSM)

@robodoo rebase-merge

**Review by rdeodoo — APPROVED** [review]

**Comment by rdeodoo** [comment]
@robodoo r+

**Comment by rdeodoo** [comment]
@robodoo retry
failed because https://github.com/odoo/odoo/pull/69888/commits/3f11eb02292e782e8659be1a4043af80ac2e2769 was merged just before this one, which needed to be adapted as website_form doesn't exists anymore :/

**Comment by rdeodoo** [comment]
@robodoo r+

File: addons/website_crm/views/website_crm_templates.xml
Comment (hoangtiendung070797) [inline addons/website_crm/views/website_crm_templates.xml:21]: @tsm-odoo  Unable to generate Lead from Contact US on Website because attribute always is `data-model_name="mail.mail". Did you miss it?

Comment (tsm-odoo) [reply addons/website_crm/views/website_crm_templates.xml:21]: Hello, just tested it on master and everything seems to work as expected. Which Odoo version are you using ? Could you please provide the steps to reproduce this issue ?

Comment (qsm-odoo) [reply addons/website_crm/views/website_crm_templates.xml:21]: @hoangtiendung070797 Did you enter edit mode, clicked on the form and changed the action option to "Create an Opportunity" ?
image

Comment (hoangtiendung070797) [reply addons/website_crm/views/website_crm_templates.xml:21]: @qsm-odoo Okay, Thank you very much. I use Odoo 15.0 and didn't see it there
cc @tsm-odoo

Comment (JKE-be) [reply addons/website_crm/views/website_crm_templates.xml:21]: @qsm-odoo does it make sense to keep this view with customize_show ? not sure...
I take a few time to understand why we got this option on the right panel, and nobody that can't read the code, could understand it imo. Maybe it could be in website module ?

HEre a good example of customer feedbacks: https://www.odoo.com/forum/website-builder-12/pass-default-values-for-form-fields-in-url-parameters-205289


//cc @pierre-pde

Comment (rdeodoo) [reply addons/website_crm/views/website_crm_templates.xml:21]: Note that we have a task to re-enable the auto form target switch when installing website_crm (not only?): https://www.odoo.com/web#id=2719426&menu_id=4720&cids=1&action=327&model=project.task&view_type=form
But I couldn't come with a clean solution yet (didn't check a second time since, wasn't a priority), I made some feedback about it on this task which was created later: https://www.odoo.com/web#id=2779452&cids=1&model=project.task&view_type=form.

In a nutshell, it is not easy in python to alter the form to send lead automatically when the module is installed as there is a lot of stuff involves which in the form transformation, which is done in JS.

Comment (JKE-be) [reply addons/website_crm/views/website_crm_templates.xml:21]: It was the tradeoff that FP decided in the past.
Nothing shocking imo, just need to know that it has been changed for old user.
For a new user, it it maybe even more logic that install a module add a feature, but don't change without warning your contact us form that you have maybe customized previously...

So, imho, remove the customize_show (before v16 freeze) if none interest to be able to toggle it, because it is confusing. If you see a good reason to keep it, I'm curious to know/understand why/which purpose...

And for the rest, to debate later in a low priority task if we really want to restore previous behaviour...

Comment (rdeodoo) [reply addons/website_crm/views/website_crm_templates.xml:21]: It is already removed for v16 with the frontend > backend task :+1: https://github.com/odoo/odoo/blob/4d3b26b5808320b12445735dd15dd24fc991c8e1/addons/website_crm/views/website_templates_contactus.xml#L4

About the debate, as there is no clean way to do it now anyway, that's not something we will be able to change

Comment (qsm-odoo) [reply addons/website_crm/views/website_crm_templates.xml:21]: @JKE-be Indeed the customize_show is just there by mistake, we'll check how to remove it in stable. For master, the customize_show="True" was removed indeed but converted as an option... also by mistake. We'll remove that with the forward-port of the stable fix.

**Pull request closed without merging**
